$MirOS: ports/emulators/qemu/snapshot/patches/patch-configure,v 1.1 2008/05/07 15:53:09 tg Exp $
$OpenBSD: patch-configure,v 1.10 2008/04/28 22:52:38 todd Exp $
--- configure.orig	Wed Sep 23 19:01:07 2009
+++ configure	Sat Oct 17 21:21:23 2009
@@ -25,11 +25,11 @@ prefix=""
 interp_prefix="/usr/gnemul/qemu-%M"
 static="no"
 cross_prefix=""
-cc="gcc"
+cc="${CC:-cc}"
 audio_drv_list=""
 audio_card_list="ac97 es1370 sb16"
 audio_possible_cards="ac97 es1370 sb16 cs4231a adlib gus"
-host_cc="gcc"
+host_cc="${CC:-cc}"
 ar="ar"
 make="make"
 install="install"
@@ -155,6 +155,9 @@ case "$cpu" in
   sparc64)
     cpu="sparc64"
   ;;
+  armish|zaurus)
+    cpu="arm"
+  ;;
   *)
     cpu="unknown"
   ;;
@@ -268,6 +271,10 @@ openbsd="yes"
 audio_drv_list="oss"
 audio_possible_drivers="oss sdl esd"
 oss_lib="-lossaudio"
+if [ "$cpu" = "i386" ] ; then
+    kqemu="yes"
+fi
+aio="no"
 ;;
 Darwin)
 bsd="yes"
@@ -511,7 +518,7 @@ for opt do
 done
 
 # default flags for all hosts
-CFLAGS="$CFLAGS -g -fno-strict-aliasing"
+CFLAGS="$CPPFLAGS $CFLAGS -g -fno-strict-aliasing"
 if test "$debug" = "no" ; then
   CFLAGS="$CFLAGS -O2"
 fi
@@ -877,7 +884,7 @@ cat > $TMPC << EOF
 #undef main /* We don't want SDL to override our main() */
 int main( void ) { return SDL_Init (SDL_INIT_VIDEO); }
 EOF
-    if $cc $ARCH_CFLAGS -o $TMPE ${OS_CFLAGS} `$sdl_config --cflags 2> /dev/null` $TMPC `$sdl_config --libs 2> /dev/null` > $TMPSDLLOG 2>&1 ; then
+    if $cc $ARCH_CFLAGS -o $TMPE ${CFLAGS} ${OS_CFLAGS} `$sdl_config --cflags 2> /dev/null` $TMPC `$sdl_config --libs 2> /dev/null` > $TMPSDLLOG 2>&1 ; then
         _sdlversion=`$sdl_config --version | sed 's/[^0-9]//g'`
         if test "$_sdlversion" -lt 121 ; then
             sdl_too_old=yes
@@ -1384,7 +1391,7 @@ else
   if test -z "$prefix" ; then
       prefix="/usr/local"
   fi
-  mansuffix="/share/man"
+  mansuffix="/man"
   datasuffix="/share/qemu"
   docsuffix="/share/doc/qemu"
   binsuffix="/bin"
@@ -1417,9 +1424,7 @@ if test "$darwin" = "yes" ; then
     echo "Cocoa support     $cocoa"
 fi
 echo "SDL support       $sdl"
-if test "$sdl" != "no" ; then
-    echo "SDL static link   $sdl_static"
-fi
+echo "SDL static link   $sdl_static"
 echo "curses support    $curses"
 echo "curl support      $curl"
 echo "mingw32 support   $mingw32"
@@ -1553,7 +1558,9 @@ EOF
 fi
 
 if [ "$openbsd" = "yes" ] ; then
+  echo "#ifndef ENOTSUP" >>$config_host_h
   echo "#define ENOTSUP 4096" >> $config_host_h
+  echo "#endif" >>$config_host_h
 fi
 
 if test "$darwin" = "yes" ; then
@@ -1802,6 +1809,8 @@ if test `expr "$target_list" : ".*softmm
   tools="qemu-img\$(EXESUF) $tools"
   if [ "$linux" = "yes" ] ; then
       tools="qemu-nbd\$(EXESUF) qemu-io\$(EXESUF) $tools"
+  elif [ "$openbsd" = "yes" ] ; then
+      tools="qemu-nbd\$(EXESUF) $tools"
   elif test "$mingw32" = "yes" ; then
       tools="qemu-io\$(EXESUF) $tools"
   fi
